File: app/src/main/java/br/com/example/meuprimeiroexemplo/activity/CommentsActivity.java
package br.com.example.meuprimeiroexemplo.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.Switch;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import br.com.example.meuprimeiroexemplo.R;
import br.com.example.meuprimeiroexemplo.adapter.CommentsAdapter;
import br.com.example.meuprimeiroexemplo.debug.DebugActivity;
import br.com.example.meuprimeiroexemplo.model.Comments;
import br.com.example.meuprimeiroexemplo.model.Post;

public class CommentsActivity extends DebugActivity {

    EditText txtPostId, txtNome, txtEmail, txtBody;
    ListView listViewPost;
    final List<HashMap<String, String>> lista = new ArrayList<>();
    final List<Comments> comments = new ArrayList<>();
    final List<Post> postagens = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_comments);
    }

    public void adicionarPost(View view) {
        //Entrada
        txtPostId = findViewById(R.id.txtPostId);
        txtNome = findViewById(R.id.txtNome);
        txtEmail = findViewById(R.id.txtEmail);
        txtBody = findViewById(R.id.txtBody);

        //Processamento
        String postId, nome, email, body;

        postId = txtPostId.getText().toString();
        nome = txtNome.getText().toString();
        email = txtEmail.getText().toString();
        body = txtBody.getText().toString();

        Switch swithc = findViewById(R.id.switch2);
        if (swithc.isChecked()) {
            System.out.println("base sendo usado");
            baseAdapter(postId, nome, email, body);
        } else {
            System.out.println("simple sendo usado");
            simpleAdapter(postId, nome, email, body);
        }
    }

    //método
    private void baseAdapter(String postId, String nome, String email, String body) {

        preencherObjetoLista(postId, nome, email, body);

        listViewPost = findViewById(R.id.listViewPost2);

        CommentsAdapter postAdapter = new CommentsAdapter(this, comments);

        listViewPost.setAdapter(postAdapter);
    }

    private void preencherObjetoLista(String postId, String nome, String email, String body) {
        try {
            Integer idConvertido = Integer.parseInt(postId);
            Comments post = Comments.builder().postId(idConvertido).nome(nome).email(email).body(body).build();

            comments.add(post);
        } catch (Exception e) {
            Toast.makeText(this, "-- Erro --" + e.getMessage(), Toast.LENGTH_LONG).show();
        }
    }

    private void simpleAdapter(String postId, String nome, String email, String body) {
        //postId,nome, email, body
        HashMap<String, String> map = new HashMap<>();
        map.put("postId", postId);
        map.put("nome", nome);
        map.put("email", email);
        map.put("body", body);

        lista.add(map);

        String[] chaves = {"postID", "nome", "email", "body"}; //chaves do map
        int[] vaiPara = {R.id.txtItemPostId, R.id.txtItemNome, R.id.txtItemEmail, R.id.txtItemBody2};//ids do layout do tipo "Item

        SimpleAdapter simpleAdapter = new SimpleAdapter(this, lista, R.layout.comments_item_post, chaves, vaiPara);

        listViewPost = findViewById(R.id.listViewPost2);
        listViewPost.setAdapter(simpleAdapter);
    }

    @Override
    public String toString() {
        return "CommentsActivity{" +
                "txtPostId=" + txtPostId +
                ", txtNome=" + txtNome +
                ", txtEmail=" + txtEmail +
                ", txtBody=" + txtBody +
                ", listViewPost=" + listViewPost +
                ", lista=" + lista +
                ", comments=" + comments +
                ", postagens=" + postagens +
                '}';
    }
}

File: app/src/main/java/br/com/example/meuprimeiroexemplo/activity/CommentsActivity.kt
package br.com.example.meuprimeiroexemplo.activity

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import br.com.example.meuprimeiroexemplo.R
import br.com.example.meuprimeiroexemplo.adapters.recyclerview.CommentsRecyclerAdapter
import br.com.example.meuprimeiroexemplo.bootstrap.APIClient
import br.com.example.meuprimeiroexemplo.debug.DebugActivity
import br.com.example.meuprimeiroexemplo.model.Comments
import br.com.example.meuprimeiroexemplo.resource.CommentsResource
import com.google.android.gms.ads.AdListener
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.LoadAdError
import com.google.android.gms.ads.MobileAds
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

open class CommentsActivity : DebugActivity() {
    private val comments: MutableList<Comments> = ArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_comments)

        //AdMob ;D

        MobileAds.initialize(applicationContext) {}
        val mAdView = findViewById<AdView>(R.id.adView2)

        val adRequest = AdRequest.Builder().build()
        mAdView.loadAd(adRequest)

        mAdView.adListener = object : AdListener() {
            override fun onAdLoaded() {
                Toast.makeText(applicationContext, "Cara, o Ad carregou ;D", Toast.LENGTH_SHORT)
                    .show()
                println("AD Carregou")
                Log.i("AD", "AD loaded")
            }

            override fun onAdFailedToLoad(adError: LoadAdError) {
                Toast.makeText(
                    applicationContext,
                    "Cara, o Ad NÃO carregou ;-;\n${adError.message}",
                    Toast.LENGTH_SHORT
                ).show()
                println("AD não Carregou??\n${adError.message}")
            }

            override fun onAdOpened() {
                // Code to be executed when an ad opens an overlay that
                // covers the screen.
            }

            override fun onAdClicked() {
                // Code to be executed when the user clicks on an ad.
            }

            override fun onAdClosed() {
                // Code to be executed when the user is about to return
                // to the app after tapping on an ad.
            }
        }
    }

    fun addComments(view: View?) {
        //Entrada
        try {
            val retrofit = APIClient.client
            val commentsResource = retrofit!!.create(CommentsResource::class.java)
            val chamada = commentsResource.get()

            chamada.enqueue(object : Callback<List<Comments>> {
                override fun onResponse(
                    call: Call<List<Comments>>,
                    response: Response<List<Comments>>
                ) {
                    Toast.makeText(applicationContext, "deu certo mano...", Toast.LENGTH_SHORT)
                        .show()
                    val postagens = response.body()
                    if (postagens != null) {
                        for (p in postagens) {
                            baseAdapter(p.postId, p.id, p.name, p.email, p.body)
                        }
                    }
                }

                override fun onFailure(call: Call<List<Comments>>, t: Throwable) {
                    Toast.makeText(
                        applicationContext,
                        "deu certo mano... mas tá sem net kk",
                        Toast.LENGTH_LONG
                    ).show()
                }

            })
            /*val b = findViewById<Button>(R.id.btnAddPost)
            b.isClickable = false
            if (!b.isClickable) {
                Toast.makeText(applicationContext, "Você não pode mais clicar!",
                        Toast.LENGTH_LONG).show()
            }*/
        } catch (e: RuntimeException) {
            Toast.makeText(
                applicationContext,
                "Deu ruim meu amigo...Erro-- ${e.message}",
                Toast.LENGTH_LONG
            ).show()
            println("\n\n\nLinha abaixo é de erro!!\n\n\n${e.message}")
        }
    }

    //método
    protected fun baseAdapter(postId: Int, id: Int, nome: String, email: String, body: String) {
        try {
            preencherObjetoLista(postId, id, nome, email, body)

            val recyclerViewPost = findViewById<RecyclerView>(R.id.listViewComments)
            val commentsRecyclerAdapter = CommentsRecyclerAdapter(applicationContext, comments)
            recyclerViewPost.adapter = commentsRecyclerAdapter
        } catch (e: RuntimeException) {
            Toast.makeText(
                applicationContext,
                "Não conseguiu usar o CommentsAdapter...${e.message}",
                Toast.LENGTH_LONG
            ).show()
        }
    }

    private fun preencherObjetoLista(
        postId: Int,
        id: Int,
        name: String,
        email: String,
        body: String
    ) {
        try {
            val comment = Comments(id, postId, name, email, body)
            comments.add(comment)
        } catch (e: RuntimeException) {
            Log.i("prencherObjetoLista\n\n\"", e.message ?: return)
            Toast.makeText(
                applicationContext, " -- Erro -- ${e.message}",
                Toast.LENGTH_LONG
            ).show()
        }
    }
}